libs/community/langchain_community/vectorstores/aerospike.py

```diff
@@ -0,0 +1,597 @@
+import warnings
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    TypeVar,
+    Union,
+)
+
+import numpy as np
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+from langchain_core.vectorstores import VectorStore
+
+from langchain_community.vectorstores.utils import (
+    DistanceStrategy,
+    maximal_marginal_relevance,
+)
+
+if TYPE_CHECKING:
+    from aerospike_vector_search import Client
+    from aerospike_vector_search.types import Neighbor, VectorDistanceMetric
+
+logger = logging.getLogger(__name__)
+
+
+def _import_aerospike() -> Any:
+    try:
+        from aerospike_vector_search import Client
+    except ImportError as e:
+        raise ImportError(
+            "Could not import aerospike_vector_search python package. "
+            "Please install it with `pip install aerospike_vector`."
+        ) from e
+    return Client
+
+
+AVST = TypeVar("AVST", bound="Aerospike")
+
+
+class Aerospike(VectorStore):
+    """`Aerospike` vector store.
+
+    To use, you should have the ``aerospike_vector_search`` python package installed.
+    """
+
+    def __init__(
+        self,
+        client: Client,
+        embedding: Union[Embeddings, Callable],
+        namespace: str,
+        index_name: Optional[str] = None,
+        vector_key: str = "_vector",
+        text_key: str = "_text",
+        id_key: str = "_id",
+        set_name: Optional[str] = None,
+        distance_strategy: Optional[
+            Union[DistanceStrategy, VectorDistanceMetric]
+        ] = DistanceStrategy.EUCLIDEAN_DISTANCE,
+    ):
+        """Initialize with Aerospike client.
+
+        Args:
+            client: Aerospike client.
+            embedding: Embeddings object or Callable (deprecated) to embed text.
+            namespace: Namespace to use for storing vectors. This should match
+            index_name: Name of the index previously created in Aerospike. This
+            vector_key: Key to use for vector in metadata. This should match the
+                key used during index creation.
+            text_key: Key to use for text in metadata.
+            id_key: Key to use for id in metadata.
+            set_name: Default set name to use for storing vectors.
+            distance_strategy: Distance strategy to use for similarity search
+                This should match the distance strategy used during index creation.
+        """
+
+        aerospike = _import_aerospike()
+
+        if not isinstance(embedding, Embeddings):
+            warnings.warn(
+                "Passing in `embedding` as a Callable is deprecated. Please pass in an"
+                " Embeddings object instead."
+            )
+
+        if not isinstance(client, aerospike):
+            raise ValueError(
+                f"client should be an instance of aerospike_vector_search.Client, "
+                f"got {type(client)}"
+            )
```

Comment:
**Issue**: The check for the 'client' instance type incorrectly uses the 'aerospike' variable which is not defined in this scope. This results in a NameError at runtime if this path is executed. <br> **Fix**: Correct the instance check to use the 'Client' class directly imported from 'aerospike_vector_search'. <br> **Code Suggestion**: 
 ```
 -        if not isinstance(client, aerospike):
 +        if not isinstance(client, Client):
 ```